src/format.rs

```diff
@@ -176,7 +177,12 @@ async fn run(
                     end_byte -= 3
                 }
 
-                content = String::from(&src[range.start_byte..end_byte]);
+                // Workaround for rst last index
+                if parser == "restructuredtext" {
```

Comment:
Instead of hard-coding exceptions here, I'd rather use treesitter directives.

Neovim has an offset one that covers this case. It shouldn't be that much more complicated to write, and it's a lot cleaner and reusable.

https://github.com/neovim/neovim/blob/001f19de2880192006eda139aa07c62a7f481422/runtime/lua/vim/treesitter/query.lua#L342

Then you can add this directly into the query, like `(#offset! @content 0 0 1 0)`

Reply:
That looks cleaner indeed.
I have some implementation that works, but I guess it needs some cleaning up.

Reply:
Yes :+1: looks good